packages/devtools_app/lib/src/charts/treemap.dart

```diff
@@ -405,31 +402,89 @@ class Treemap extends StatelessWidget {
           ],
         ),
       );
+    } else {
+      return Column(
+        children: [
+          if (levelsVisible == 2 && height > minHeightToDisplayTitleText)
+            buildTitleText(),
+          Expanded(
+            child: buildSelectable(
+              child: Container(
+                decoration: BoxDecoration(
+                  color: mainUiColor,
+                  border: Border.all(color: Colors.black54),
+                ),
+                child: Center(
+                  child: height > minHeightToDisplayCellText
+                      ? buildNameAndSizeText(
+                          fontColor: Colors.black,
+                          oneline: false,
+                        )
+                      : const SizedBox(),
+                ),
+              ),
+            ),
+          ),
+        ],
+      );
+    }
+  }
+
+  Widget buildTitleText() {
+    if (levelsVisible == 2) {
+      final pathFromRoot = rootNode.pathFromRoot();
+      return Container(
+        height: treeMapHeaderHeight,
+        child: ListView.separated(
+          shrinkWrap: true,
+          scrollDirection: Axis.horizontal,
+          separatorBuilder: (context, index) {
+            return const Text(' > ');
+          },
+          itemCount: pathFromRoot.length,
+          itemBuilder: (BuildContext context, int index) {
+            return buildSelectable(
+              child: Text(
+                index < pathFromRoot.length - 1
+                    ? pathFromRoot[index].name
+                    : pathFromRoot[index].displayText(),
+              ),
+              newRoot: pathFromRoot[index],
+            );
+          },
+        ),
+      );
     } else {
       return buildSelectable(
         child: Container(
+          width: double.infinity,
           decoration: BoxDecoration(
-            color: mainUiColor,
-            border: Border.all(),
+            border: Border.all(color: Colors.black54),
           ),
-          child: Center(
-            child: buildNameAndSizeText(
-              fontColor: Colors.black,
-              oneline: false,
-            ),
+          child: buildNameAndSizeText(
+            fontColor: Colors.white,
```

Comment:
this won't work for light theme. Use Theme.of(context).textTheme to grab a color